### DISCUSSION TITLE
ROM from non-merged set still needs BIOS in Flycast?

### DISCUSSION THREAD

**Discussion by kathyrollo** [flyinghead/flycast#1251] [Q&A] [answered]
Hi @flyinghead

I obtained `dolphin.zip` (Dolphin Blue) from the non-merged set, but Flycast still looks for `awbios.zip` to run it. From my understanding, files from the non-merged set no longer need other files to run as everything needed is already included in the archive.

Any reason why Flycast (standalone version) still looks for the BIOS?

![Screenshot 2023-10-19 162549](https://github.com/flyinghead/flycast/assets/28589393/37f2622e-5168-417c-95c5-8e74eb96a698)

Quoted from PD:
> non-merged:
A non-merged set is one that contains absolutely everything necessary for a given game to run in one ZIP file.
Remark: Even though Non-Merged games -normally- DO NOT include BIOS or DEVICE files within the game, in the case of the PD non-merged set we DO include these within the game, which ensures that every single game can be copied and ran as individual game without the need of any extra files.
This makes a set like this ideal for splitting up on a per-game-basis. The non-merged set is ideal for those people that work on Arcade PCBs as ALL roms/devices/bios files are contained within the game. This set is also great for those that for instance create their own arcade cabinets and want to copy only very specific games to their PC/Rapsberry/Other, the game.zip file contain all the files needed, no more searching for the dependent parent files, BIOS files, device files - just copy galaga.zip and you are set.
Pick individual games from this set if you are not intending to download a complete ROM set.

**Comment by cinnamonmatexfce** [comment]
I think _type_ of sets (merge, split, unmerge) only applies to MAME.
In its code, MAME can recursively look for _set_'s required roms based on their relationships. 

For what I understand, OTOH Flycast simply look for a list of needed roms for every sets, with no relationships declared.
Please have a look here: https://github.com/flyinghead/flycast/blob/master/core/hw/naomi/naomi_roms.cpp

**Comment by flyinghead** [comment, marked as answer]
Flycast does take roms relationship into account. And I believe it loads the bios files from the rom if included.
However the case where the bios file doesn't exist at all isn't correctly handled at the moment.

**Reply by kathyrollo** [reply]
Thanks. Guess I'll continue to pick ROMs from the non-merged set (as it serves purpose for other emulators) and keep the BIOS files for Flycast.
